### DISCUSSION TITLE
disable (and scrub existing) history permanently

### DISCUSSION THREAD

**Discussion by mkvlrn-cm42** [lzybkr/PSReadLine#3733] [Q&A] [answered]
**TL;DR: how to disable history?**

hi, I'm making a move from linux to windows development and I'm using pwsh with oh-my-posh and my prompt looks absolutely fabulous!

just *one* thing that irks me is the history that tries to give me suggestions when I'm typing.

at first I thought it was something to do with oh-my-posh itself, but after searching a bit it seems the functionality comes from PSReadLine.

I just need history and the attempt to give me suggestions from it gone, but I don't want to disable auto completing, such as pressing tab after `Get-` to get suggestions *there*. does that make sense?

**Comment by mkvlrn-cm42** [comment, marked as answer]
welp, I should have done a bit more research! found the answer a bit after posting this, so I'll just share anyway:

~~https://www.blackhillsinfosec.com/new-powershell-history-defense-evasion-technique/~~

~~`Remove-Item (Get-PSReadlineOption).HistorySavePath` to delete history~~
~~`Set-PSReadLineOption -AddToHistoryHandler { return $false }` in your profile to disable~~

real solution is to just disable predictions. duh.
`Set-PSReadLineOption -PredictionSource None` in the profile. k, I'm done editing now.

cheers!
